include/RAJA/pattern/reduce.hpp

```diff
@@ -64,6 +65,67 @@
 namespace RAJA
 {
 
+namespace reduce
```

Comment:
Would assembly output be enough to quantify the overhead vs manual/macro alternative?